packages/@aws-cdk/aws-cloudfront/README.md

```diff
@@ -103,3 +114,118 @@ const distribution = new CloudFrontWebDistribution(this, 'MyDistribution', {
     ]
 });
 ```
+
+## Distribution API - Experimental
+
+![cdk-constructs: Experimental](https://img.shields.io/badge/cdk--constructs-experimental-important.svg?style=for-the-badge)
+
+In addition to the APIs listed above, a new construct (`Distribution`) is under active development as a newer, friendlier way to define CloudFront
+distributions.
+
+### Creating a distribution
+
+CloudFront distributions deliver your content from one or more origins; an origin is the location where you store the original version of your
+content. Origins can be created from S3 buckets or a custom origin (HTTP server). Each distribution has a default behavior which applies to all
+requests to that distribution, and routes requests to a primary origin.
+
+#### From an S3 Bucket
+
+An S3 bucket can be added as an origin. If the bucket is configured as a website endpoint, the distribution can use S3 redirects and S3 custom error
+documents.
+
+```ts
+import * as cloudfront from '@aws-cdk/aws-cloudfront';
+
+// Creates a distribution for a S3 bucket.
+const myBucket = new s3.Bucket(...);
+new cloudfront.Distribution(this, 'myDist', {
+  defaultBehavior: { origin: cloudfront.Origin.fromBucket(myBucket) },
+});
+
+// Equivalent to the above
+const myBucket = new s3.Bucket(...);
+cloudfront.Distribution.forBucket(this, 'myDist', myBucket);
+
+// Creates a distribution for a S3 bucket that has been configured for website hosting.
+const myWebsiteBucket = new s3.Bucket(...);
+new cloudfront.Distribution(this, 'myDist', {
+  defaultBehavior: { origin: cloudfront.Origin.fromWebsiteBucket(myBucket) },
```

Comment:
Can we add info the s3.Bucket and identify that it is a website bucket automatically? What's the benefit of the additional API. Cognitive load seems unwarranted.

Reply:
The info exists on `s3.Bucket`, but not `s3.IBucket`. A bucket configured for website hosting will have some combination of 4 properties (`websiteRoutingRules`, `websiteRedirect`, `websiteIndexDocument`, or `websiteErrorDocument`) set. We could move these four properties to IBucket (and BucketAttributes) as optional and then support the appropriate behavior here _if_ those properties were set, or we could create a meta-property (`isWebsiteBucket`) that is only used for this (and has to validate with the Bucket properties), or we can do what I've done here. The latter seemed least intrusive and makes it explicit that behaviors are different between the two options. As more properties are added to this, things like defaultRootIndex should be set differently between the two. 

My concern with moving those props to IBucket would be that existing code that imports buckets won't set them, and it won't be obvious from usage that because those properties aren't set, the Distribution is doing the wrong thing. I opted for the more explicit choice instead.

@eladb - Would you argue for one of the other above options (e.g., moving the website properties to IBucket)?

Reply:
`isWebsite` in `IBucket` sounds like a good option. If people use `Bucket.fromAttributes` they will be able to explicitly specify it if they want and the reasonable default would be `false`.

Reply:
Ok, done.